fix: change strides parameter to be pointer to const

PraveshKunwar opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: na
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: na
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: missing_dependencies
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

Resolves #6604 

## Description

> What is the purpose of this pull request?

This pull request:

-   Changes parameter (strides) to be declared as pointer to const.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #6604 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

PraveshKunwar committed:
fix: change strides parameter to be pointer to const

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/ndarray/base/assert/is-column-major/src/main.c:
```diff
@@ -40,7 +40,7 @@
 * // returns 1
 */
-int8_t stdlib_ndarray_is_column_major( int64_t ndims, int64_t *strides ) {
+int8_t stdlib_ndarray_is_column_major( int64_t ndims, const int64_t *strides ) {
```
```suggestion
int8_t stdlib_ndarray_is_column_major( const int64_t ndims, const int64_t *strides ) {
```

kgryte commented:
@PraveshKunwar Would you also mind updating the corresponding header file in the same package to match the updated signature?

PraveshKunwar committed:
fix: changed ndims to be const

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

PraveshKunwar commented:
@kgryte Yep, just fixed it!

kgryte commented on lib/node_modules/@stdlib/ndarray/base/assert/is-column-major/src/main.c:
```diff
@@ -40,7 +40,7 @@
 * // returns 1
 */
-int8_t stdlib_ndarray_is_column_major( int64_t ndims, int64_t *strides ) {
+int8_t stdlib_ndarray_is_column_major(const int64_t ndims, const int64_t *strides ) {
```
```suggestion
int8_t stdlib_ndarray_is_column_major( const int64_t ndims, const int64_t *strides ) {
```

kgryte committed:
style: add missing space

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
> Would you also mind updating the corresponding header file in the same package to match the updated signature?

@PraveshKunwar This still needs to be done. This PR should contain changes to two files: 1) the source file and 2) the associated header file, which can be found in the same package in the `include` directory.

PraveshKunwar committed:
fix: changed associated header file

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

PraveshKunwar committed:
Merge branch 'stdlib-js:develop' into praveshk/c-errors

PraveshKunwar commented:
@kgryte ok, I believe everything should be ok now

kgryte reviewed (approved):
LGTM
